File: python/pycommons/math/angles.py
"""Calculations involving angles"""

import math


def getWrappedAngleDegrees(current: float, desired: float) -> float:
    """Given any two angles, get the closest distance between them

    Args:
        current (float): Current / starting angle
        desired (float): Desired / end angle

    Returns:
        float: Closest distance
    """

    # This is either the distance or 360 - distance
    phi = math.abs(desired - current) % 360
    distance = 360 - phi if phi > 180 else phi

    # Determine the sign (is the difference positive of negative)
    sign = 1 if (desired - current >= 0 and desired - current <=
                 180) or (desired - current <= -180 and desired - current >= -360) else -1

    # Return the final difference
    return distance * sign


def getWrappedErrorRadians(current: float, desired: float) -> float:
    """Given any two angles, get the closest distance between them

    Args:
        current (float): Current / starting angle
        desired (float): Desired / end angle

    Returns:
        float: Closest distance
    """

    return math.radians(getWrappedAngleDegrees(math.degrees(current), math.degrees(desired)))

File: java/src/main/java/ca/retrylife/commons/math/Angles.java
package ca.retrylife.commons.math;

/**
 * Calculations involving angles
 */
public class Angles {

    /**
     * Given any two angles, get the closest distance between them
     * 
     * @param current Current / starting angle
     * @param desired Desired / end angle
     * @return Closest distance
     */
    public static double getWrappedErrorDegrees(double current, double desired) {
        double phi = Math.abs(desired - current) % 360; // This is either the distance or 360 - distance
        double distance = phi > 180 ? 360 - phi : phi;

        // Determine the sign (is the difference positive of negative)
        int sign = (desired - current >= 0 && desired - current <= 180)
                || (desired - current <= -180 && desired - current >= -360) ? 1 : -1;

        // Return the final difference
        return distance * sign;
    }

    /**
     * Given any two angles, get the closest distance between them
     * 
     * @param current Current / starting angle
     * @param desired Desired / end angle
     * @return Closest distance
     */
    public static double getWrappedErrorRadians(double current, double desired) {
        return Math.toRadians(getWrappedErrorDegrees(Math.toDegrees(current), Math.toDegrees(desired)))
    }

}